CACHING.md

```diff
@@ -0,0 +1,351 @@
+        path/to/dependencies
+        some/other/dependencies
+      key: ${{ runner.os }}-${{ hashFiles('**/lockfiles') }}
+```
+
+The [GitHub Context](https://docs.github.com/en/actions/learn-github-actions/contexts#github-context) can be used to create keys using the workflows metadata.
+
+## Restoring Cache
+
+### Know paths better
+
+While setting paths for caching dependencies it is important to give correct path depending on the image you are using and if you are running the action inside or outside the container. Below are some paths one should keep in mind while writing their workflow files.
+
+#### Ubuntu Paths
+
+```yaml
+process.env['RUNNER_TEMP']=/home/runner/work/_temp 
+process.cwd()=/home/runner/work/repo/repo
+Home directory = /home/runner
+${{ github.workspace }} = /home/runner/work/repo/repo
+```
+
+#### Windows Paths
+
+```yaml
+process.env['RUNNER_TEMP']=D:\a\_temp 
+process.cwd()=D:\a\repo\repo 
+Home directory = C:\Users\runneradmin
+${{ github.workspace }} = D:\a\repo\repo
+```
+
+#### MacOS Paths
+
+```yaml
+process.env['RUNNER_TEMP']=/Users/runner/work/_temp 
+process.cwd()=/Users/runner/work/repo/repo 
+Home directory = /Users/runner
+${{ github.workspace }} = /Users/runner/work/repo/repo
+```
+
+Where:
+Home directory  = `~/`.
+cwd() = Current working directory where code executes by default.
+RUNNER_TEMP = Environment variable defined for temporary storage location.
+
+### Make cache read only / Reuse cache from centralized job
+
+In case you are using a centralized job to create and save your cache that can be reused by other jobs in your repository, this action will take care of your restore only needs and make the cache read-only.
+
+```yaml
+steps:
+  - uses: actions/checkout@v3
+
+  - uses: actions/cache/restore@v3
+    id: cache
+    with:
+      path: path/to/dependencies
+      key: ${{ runner.os }}-${{ hashFiles('**/lockfiles') }}
+
+  - name: Install Dependencies
+    if: steps.cache.outputs.cache-hit != 'true'
+    run: /install.sh
+
+  - name: Build
+    run: /build.sh
+
+  - name: Publish package to public
+    run: /publish.sh
+```
+
+### Failing/Exiting the workflow if cache with exact key is not found
+
+You can use the output of this action to exit the workflow on cache miss. This way you can restrict your workflow to only initiate the build when `cache-hit` occurs, in other words, cache with exact key is found.
+
+```yaml
+steps:
+  - uses: actions/checkout@v3
+
+  - uses: actions/cache/restore@v3
+    id: cache
+    with:
+      path: path/to/dependencies
+      key: ${{ runner.os }}-${{ hashFiles('**/lockfiles') }}
+
+  - name: Check cache hit
+    if: steps.cache.outputs.cache-hit != 'true'
+    run: exit 1
+
+  - name: Build
+    run: /build.sh
+```
+
+## Saving cache
+
+### Reusing primary key from restore cache as input to save action
+
+If you want to avoid re-writing the cache key again in `save` action, the outputs from `restore` action can be used as input to the `restore` action.
```

Comment:
```suggestion
If you want to avoid re-computing the cache key again in `save` action, the outputs from `restore` action can be used as input to the `restore` action.
```